test/Behodler.test.js

```diff
@@ -1,102 +1,153 @@
+        await this.lachesis.measure(this.regularToken.address, true, false)
+        await this.lachesis.measure(this.burnableToken.address, true, true)
+
+        await this.lachesis.setBehodler(this.behodler.address)
+        await this.lachesis.updateBehodler(this.regularToken.address)
+        await this.lachesis.updateBehodler(this.burnableToken.address)
     })
 
     it('adding burnable token as liquidity in 2 batches generates the correct volume of Scarcity', async function () {
         //ADD 1 FINNEY WHEN BEHODLER BALANCE OF TOKEN ZERO
         const originalBalance = 2000000n * TEN
         const expectedBalanceAfter = originalBalance - FINNEY * 2n
 
-        const scarcitySupplyBefore = await bigNum.BNtoBigInt(this.behodler.totalSupply.call())
+        const scarcitySupplyBefore = await bigNum.BNtoBigInt(this.behodler.totalSupply())
         expect(scarcitySupplyBefore).to.equal(0n)
 
-        await this.burnableToken.approve(this.behodler.address, originalBalance, { from: trader1 })
-        await this.behodler.addLiquidity(this.burnableToken.address, FINNEY * 2n, { from: trader1 })
-        const tokenBalanceOfUser = await bigNum.BNtoBigInt(this.burnableToken.balanceOf.call(trader1))
-        expect(tokenBalanceOfUser).to.equal(expectedBalanceAfter)
+        await this.burnableToken.connect(trader1).approve(this.behodler.address, originalBalance)
+        await this.behodler.connect(trader1).addLiquidity(this.burnableToken.address, FINNEY * 2n)
 
-        const scarcityBalance = (await this.behodler.balanceOf.call(trader1)).toString()
+        const tokenBalanceOfUser = await bigNum.BNtoBigInt(this.burnableToken.balanceOf(trader1.address))
+        
+        assertBNequal(tokenBalanceOfUser, expectedBalanceAfter)
+
+        const scarcityBalance = (await this.behodler.balanceOf(trader1.address)).toString()
 
         const expectedScarcity = '201609232779564367355'
-        expect(scarcityBalance).to.equal(expectedScarcity)
+        assertBNequal(scarcityBalance, expectedScarcity)
 
-        const scarcitySupplyAfter = (await bigNum.BNtoBigInt(this.behodler.totalSupply.call())).toString()
-        expect(scarcitySupplyAfter).to.equal(expectedScarcity)
+        const scarcitySupplyAfter = (await bigNum.BNtoBigInt(this.behodler.totalSupply())).toString()
+        assertBNequal(scarcitySupplyAfter, expectedScarcity)
 
-        await this.behodler.addLiquidity(this.burnableToken.address, 2n * FINNEY, { from: trader1 })
+        await this.behodler.connect(trader1).addLiquidity(this.burnableToken.address, 2n * FINNEY)
         const expectedBalanceAfterSecondAdd = expectedBalanceAfter - 2n * FINNEY
-        const tokenBalanceOfUserAfterSecondAdd = await bigNum.BNtoBigInt(this.burnableToken.balanceOf.call(trader1))
-        expect(tokenBalanceOfUserAfterSecondAdd).to.equal(expectedBalanceAfterSecondAdd)
+        const tokenBalanceOfUserAfterSecondAdd = await bigNum.BNtoBigInt(this.burnableToken.balanceOf(trader1.address))
```

Comment:
const tokenBalanceOfUserAfterSecondAdd = await this.burnableToken.balanceOf(trader1.address)

+ remove bigNum.BNtoBigInt everywhere where is not needed

Reply:
removed